### PR TITLE
fix dependabot.yaml indentation

### PR DESCRIPTION
**Pull request by nawazkh**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
- We haven't seen any dependabot activities at CAPZ.
- Comparing [CAPZ's dependabot.yml ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.github/dependabot.yml) with [CAPI's dependabot.yml](https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/dependabot.yml) points to minor indentation difference b/w both the files.
- Let's try to merge this change to see if it enables Dependabot at CAPZ
  
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # -- 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by mboersma — COMMENTED** [review]
/lgtm

**Comment by CecileRobertMichon** [comment]
Looks like we already have a dependabot.yml file https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.github/dependabot.yml, is that different?

**Comment by nawazkh** [comment]
> Looks like we already have a dependabot.yml file https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.github/dependabot.yml, is that different?

Thanks for the pointing it out. I overlooked it and was heading in a different direction altogether.

Updating `dependabot-code-gen.yaml -> dependabot.yaml` is not related to get Dependabot working as [dependabot-code-gen.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.github/workflows/dependabot-code-gen.yml) is a GitHub workflow (which doesn't seem to be working either! 😕) 

I compared CAPZ's dependabot.yml with CAPI's [dependabot.yml](https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/dependabot.yml) and the only difference I found was with indentation. So let's try if this fix enables Dependabot.

Also note, CAPZ's `dependabot-code-gen` workflow would fail as we do not have any rules defined in Makefile for [`generate-modules`](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.github/workflows/dependabot-code-gen.yml#L37). I will fix it in another PR.

**Comment by CecileRobertMichon** [comment]
> Updating dependabot-code-gen.yaml -> dependabot.yaml is not related to get Dependabot working as [dependabot-code-gen.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.github/workflows/dependabot-code-gen.yml) is a GitHub workflow (which doesn't seem to be working either! 😕)

maybe that one only gets triggered once a dependabot PR gets open? Either way we'll find out soon

> I compared [CAPZ's dependabot.yml ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.github/dependabot.yml)with [CAPI's dependabot.yml](https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/dependabot.yml) and the only difference I found was with indentation. So let's try this fix if it enables Dependabot.

Great find!

> Also note, CAPZ's dependabot-code-gen workflow would fail as we do not have any rules defined in Makefile for [generate-modules](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.github/workflows/dependabot-code-gen.yml#L37). I will fix it in another PR.

👍 

/lgtm
/approve
/hold cancel

File: .github/dependabot.yml [hunk 1]
```diff
@@ -2,37 +2,35 @@
...
-    commit-message:
-      include: "scope"
-    labels:
-      - "kind/cleanup"
-      - "ok-to-test"
+# Dependencies listed in go.mod
+- package-ecosystem: "gomod"
+  directory: "/" # Location of package manifests
+  schedule:
+    interval: "weekly"
```
Comment (CecileRobertMichon) [inline .github/dependabot.yml:9]: /hold

maybe change this until we get it working for a quicker feedback loop?

Comment (nawazkh) [reply .github/dependabot.yml]: added via e0ec0e926f6bf969651685e2fff75a6f8ee394e6

**Review by mboersma — COMMENTED** [review]
/lgtm

Thanks for digging into this @nawazkh! This looks like proper YAML indentation, let's hope that wakes up the Dependabot.

**Pull request merged**